packages/providers/src.ts/base-provider.ts

```diff
@@ -623,20 +623,33 @@ export class BaseProvider extends Provider {
         }
     }
 
-    async getCode(addressOrName: string | Promise<string>, blockTag?: BlockTag | Promise<BlockTag>): Promise<string> {
-        await this.getNetwork();
-        const params = await resolveProperties({
-            address: this._getAddress(addressOrName),
-        });
-
-        const result = await this.perform("getCode", params);
+    /**
+     *  Get contract bytecode implementation, using the REST Api.
+     *  It returns the bytecode, or a default value as string.
+     *
+     * @param addressOrName The address to obtain the bytecode of
+     */
+    async getCode(addressOrName: string | Promise<string>, throwOnNonExisting?: boolean): Promise<string> {
+        if (!this._mirrorNodeUrl) logger.throwError("missing provider", Logger.errors.UNSUPPORTED_OPERATION);
```

Comment:
Lets add a method that we can reuse `checkMirrorNode` f.e and we can call it whenever we need to do this check

Reply:
added a `checkMirrorNode` method in `BaseProvider`